packages/glimmer-syntax/lib/traversal/walker.ts

```diff
@@ -1,50 +1,53 @@
-function Walker(order=undefined) {
-  this.order = order;
-  this.stack = [];
-}
-
-export default Walker;
-
-Walker.prototype.visit = function(node, callback) {
-  if (!node) {
-    return;
-  }
-
-  this.stack.push(node);
-
-  if (this.order === 'post') {
-    this.children(node, callback);
-    callback(node, this);
-  } else {
-    callback(node, this);
-    this.children(node, callback);
-  }
-
-  this.stack.pop();
-};
-
 let visitors = {
-  Program: function(walker, node, callback) {
+  Program(walker, node, callback) {
     for (let i = 0; i < node.body.length; i++) {
       walker.visit(node.body[i], callback);
     }
   },
 
-  ElementNode: function(walker, node, callback) {
+  ElementNode(walker, node, callback) {
     for (let i = 0; i < node.children.length; i++) {
       walker.visit(node.children[i], callback);
     }
   },
 
-  BlockStatement: function(walker, node, callback) {
+  BlockStatement(walker, node, callback) {
     walker.visit(node.program, callback);
     walker.visit(node.inverse, callback);
   }
 };
 
-Walker.prototype.children = function(node, callback) {
-  let visitor = visitors[node.type];
-  if (visitor) {
-    visitor(this, node, callback);
+export default class Walker {
+  private order: String | undefined;
```

Comment:
We don't have strict null check on master yet, so I don't think `| undefined` actually do anything (might in fact be illegal), but we will have it very soon. On that branch I think we can use `Option<string>` or something like that